permission_handler_apple/ios/Classes/strategies/EventPermissionStrategy.m

```diff
@@ -37,49 +37,113 @@ - (void)requestPermission:(PermissionGroup)permission completionHandler:(Permiss
+            #if PERMISSION_EVENTS_FULL_ACCESS
+            [eventStore requestFullAccessToEventsWithCompletion:^(BOOL granted, NSError *error) {
+                    if (granted) {
+                        completionHandler(PermissionStatusGranted);
+                    } else {
+                        completionHandler(PermissionStatusPermanentlyDenied);
+                    }
+                }];
+            #elif PERMISSION_EVENTS
+            [eventStore requestWriteOnlyAccessToEventsWithCompletion:^(BOOL granted, NSError *error) {
+                    if (granted) {
+                        completionHandler(PermissionStatusGranted);
+                    } else {
+                        completionHandler(PermissionStatusPermanentlyDenied);
+                    }
+                }];
+            #endif
         } else {
-            completionHandler(PermissionStatusPermanentlyDenied);
+            #if PERMISSION_REMINDERS
+            [eventStore requestFullAccessToRemindersWithCompletion:^(BOOL granted, NSError *error) {
+                                if (granted) {
+                                    completionHandler(PermissionStatusGranted);
+                                } else {
+                                    completionHandler(PermissionStatusPermanentlyDenied);
+                                }
+                   }];
+            #endif
         }
-    }];
+    } else {
+        [eventStore requestAccessToEntityType:entityType completion:^(BOOL granted, NSError *error) {
+                if (granted) {
+                    completionHandler(PermissionStatusGranted);
+                } else {
+                    completionHandler(PermissionStatusPermanentlyDenied);
+                }
+            }];
+    }
+
 }
 
 + (PermissionStatus)permissionStatus:(EKEntityType)entityType {
     EKAuthorizationStatus status = [EKEventStore authorizationStatusForEntityType:entityType];
-    
-    switch (status) {
-        case EKAuthorizationStatusNotDetermined:
-            return PermissionStatusDenied;
-        case EKAuthorizationStatusRestricted:
-            return PermissionStatusRestricted;
-        case EKAuthorizationStatusDenied:
-            return PermissionStatusPermanentlyDenied;
-        case EKAuthorizationStatusAuthorized:
-            return PermissionStatusGranted;
+
+    if (@available(iOS 17.0, *)) {
+        switch (status) {
+                case EKAuthorizationStatusNotDetermined:
+                    return PermissionStatusDenied;
+                case EKAuthorizationStatusRestricted:
+                    return PermissionStatusRestricted;
+                case EKAuthorizationStatusDenied:
+                    return PermissionStatusPermanentlyDenied;
+                case EKAuthorizationStatusFullAccess:
+                    return PermissionStatusGranted;
+                case EKAuthorizationStatusWriteOnly:
+                    #if PERMISSION_EVENTS_FULL_ACCESS
```

Comment:
Is it always either one? Because otherwise you could be missing an "else" return.

Reply:
I went with assumption that it's always either one, which was wrong, it might be both at the same time. I think the latest commit addresses this issue.